### DISCUSSION TITLE
Lossless streaming

### DISCUSSION THREAD

**Discussion by vale-max** [bluenviron/mediamtx#401] [Questions]
Hi all,
I have an input stream which is a raw PCM 48k stereo which I'd like to make available in my local LAN, so having no bandwidth restrictions I'd like to stream it in a lossless format.

If I understood correctly, rtsp does not allow lossless codecs, so I wanted to stream using hls and flac (or alac).
Is it possible to do this using rtsp-simple-server?
Or is there another way to stream this lossless with rtsp-simple-server?

**Comment by dorinclisu** [comment]
Does HLS support lossless video? As far as I can see it does seem to support lossless audio, but video must be H264 or H265.

**Reply by vale-max** [reply]
My stream is audio only, no video

**Reply by aler9** [reply]
It is technically possible to stream LPCM with RTSP, since LPCM can be wrapped inside RTP packets:

https://en.wikipedia.org/wiki/RTP_payload_formats

The problem is that neither FFmpeg nor Gstreamer support it. FFmpeg tries to instantiate a connection to the server, but it doesn't provide the stream codec (this is needed for a series of reasons), and the connection is rejected.

[a comment by an automated account is not shown]
